morphia/src/main/java/org/mongodb/morphia/DatastoreImpl.java

```diff
@@ -463,13 +399,30 @@ public void ensureIndexes(final boolean background) {
         return getByKeys(null, keys);
     }
 
+    private DBCollection getCollection(final Object obj) {
+        if (obj == null) {
+            return null;
```

Comment:
It gets called from a number of places that accept user input.  There's a decent chance nulls might make it in here.